jobqueue/server.go

```diff
@@ -233,14 +233,17 @@ type Server struct {
 	simutex            sync.RWMutex
 	krmutex            sync.RWMutex
 	ssmutex            sync.RWMutex // "server state mutex" to protect up, drain, blocking and ServerInfo.Mode
+	rpmutex            sync.Mutex   // to protect racPending and waitingReserves
 	sync.Mutex
-	sgcmutex    sync.Mutex
-	wsmutex     sync.Mutex
-	up          bool
-	drain       bool
-	blocking    bool
-	racChecking bool
-	killRunners bool
+	sgcmutex        sync.Mutex
+	wsmutex         sync.Mutex
+	up              bool
+	drain           bool
+	blocking        bool
+	racChecking     bool
+	killRunners     bool
+	racPending      bool
```

Comment:
A brief comment describing the intent behind field would be helpful (likewise for other fields, some other time)

Reply:
Agreed